spirit/comment/forms.py

```diff
@@ -138,3 +139,47 @@ def save(self):
         name = default_storage.save(name, file)
         file.url = default_storage.url(name)
         return file
+
+
+class CommentFileForm(forms.Form):
+
+    file = forms.FileField()
+
+    def __init__(self, user=None, *args, **kwargs):
+        super(CommentFileForm, self).__init__(*args, **kwargs)
+        self.user = user
+
+    def clean_file(self):
+        file = self.cleaned_data['file']
+        try:
+            file_mime = magic.from_buffer(file.read(131072), mime=True)
+        except magic.MagicException as e:
+            raise forms.ValidationError(_(e.message))
```

Comment:
gettext only works on "static" strings. I think it will just throw an error when trying to compile the translations otherwise. IMHO we should just return a generic error like "The file could not be validated" and log the error:

```python
import logging

logger = logging.getLogger('django')

# ...
def clean_file(self):
    # ...
    except magic.MagicException as e:
        logger.exception(e)
```

Reply:
That is a good idea and I didn't think about the compiling/translations aftereffect. :+1:

Reply:
Are you opposed to `logging.getLogger(__name__)`? Or did you want to set it to `logging.getLogger('django')` specifically?

Reply:
Oh, that's better! let's set it to `__name__`